### ISSUE TITLE
Incompatibility with NumPy 2.0 due to deprecated `np.sctypes`

### ISSUE THREAD

**Issue by DaniJonesOcean** [wesselb/lab#20]
#### Summary
The `lab` package currently specifies `numpy>=1.16` as a requirement, which leads to the installation of NumPy 2.0 when setting up a new environment or upgrading existing packages. Due to significant changes in NumPy 2.0, specifically the removal of `np.sctypes`, the `lab` package fails to import when used with the latest version of NumPy.

#### Environment
- **Operating System:** Linux (i.e. Red Hat Enterprise Linux release 8.8)
- **Python Version:** 3.10 
- **Installation Method:** pip/virtual environment

#### Steps to Reproduce
1. Set up a Python 3.10 environment with pip.
2. Install the `lab` package with the latest version of NumPy: `pip install 'numpy>=1.16' lab` (NumPy 2.0 will be installed).
3. Attempt to import that uses `np.sctypes`, resulting in an AttributeError (e.g. `import lab as B`)

#### Expected Behavior
The `lab` package should be able to import successfully without any AttributeError regarding `np.sctypes`.

#### Actual Behavior
An AttributeError is raised when importing the `lab` package with NumPy 2.0 installed:

```plaintext
AttributeError: `np.sctypes` was removed in the NumPy 2.0 release. Access dtypes explicitly instead.. Did you mean: 'dtypes'?
```

For example, when you try to import the [deepsensor](https://github.com/alan-turing-institute/deepsensor) package, which uses the `lab` package:
```
  File "/home/dannes/deepsensor_numpy_test/lib/python3.10/site-packages/lab/__init__.py", line 9, in <module>
    from .generic import *
  File "/home/dannes/deepsensor_numpy_test/lib/python3.10/site-packages/lab/generic.py", line 11, in <module>
    from .types import (
  File "/home/dannes/deepsensor_numpy_test/lib/python3.10/site-packages/lab/types.py", line 99, in <module>
    Int = Union[tuple([int, Dimension] + np.sctypes["int"] + np.sctypes["uint"])]
  File "/home/dannes/deepsensor_numpy_test/lib/python3.10/site-packages/numpy/__init__.py", line 397, in __getattr__
    raise AttributeError(
AttributeError: `np.sctypes` was removed in the NumPy 2.0 release. Access dtypes explicitly instead.. Did you mean: 'dtypes'?
```

#### Suggested Solution
Temporarily restrict the version of NumPy in `setup.py` to prevent installation of incompatible versions. I suggest changing the requirement in `setup.py` to:

```python
"numpy<2"
```

**Comment by DaniJonesOcean** [comment]
Posted related issue on DeepSensor repository:

https://github.com/alan-turing-institute/deepsensor/issues/118

**Comment by wesselb** [comment]
Hey @DaniJonesOcean! Thanks for opening an issue. Let me look into this. I think there should be a simple fix by changing the import. I'll get back to you shortly.

**Comment by wesselb** [comment]
@DaniJonesOcean This should be fixed in the latest release `v1.6.6`.

**Comment by DaniJonesOcean** [comment]
Great, thanks for your quick work on this, @wesselb!
